pandas/_libs/lib.pyx

```diff
@@ -2217,6 +2218,9 @@ def maybe_convert_objects(ndarray[object] objects, bint try_float=False,
                 seen.object_ = True
                 break
         elif util.is_integer_object(val):
+            if getattr(val, 'name', None) is not None and not convert_intenum:
```

Comment:
i am not sure why you need this extra argument at all?  the getattr (or a hasattr check) would prob be ok

Reply:
in fact it could be in the elif itself

Reply:
Thanks @jreback, I will take a look at this when I get a chance later this week.